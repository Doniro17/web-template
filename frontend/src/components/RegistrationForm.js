import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import Input from './Input';
import API from '../utils/API';

export default function RegistrationForm() {
  const history = useHistory();
  const [state, setState] = useState({
    login: '',
    password: '',
    email: '',
    phone: '',
    client: false,
    courier: false,
    role: [],
  });
  const { login, password, email, phone, role, client, courier } = state;

  function handleChange(event) {
    const { target } = event;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const { name } = target;
    setState((oldState) => ({
      ...oldState,
      [name]: value,
    }));
  }
  function onSend(event) {
    const loginToSystem = async () => {
      if (client) role.push('client');
      if (courier) role.push('courier');
      await API.post('users/register', {
        username: login,
        password,
        email,
        phone,
        balance: 0,
        role,
      })
        .then(() => {
          history.push('/login');
        })
        .catch((e) => {
          if (e.response.status === 400) {
            // eslint-disable-next-line no-alert
            alert('WIP');
          }
        });
    };
    loginToSystem();
    event.preventDefault();
  }
  return (
    <form onSubmit={onSend} className="login__form">
      <h2 className="form__title">Registration</h2>
      <Input
        label="Client"
        labelStyle="container"
        type="checkbox"
        value={client}
        onChange={handleChange}
        className="form__checkbox"
        name="client"
      />
      <Input
        label="Courier"
        labelStyle="container"
        type="checkbox"
        value={courier}
        onChange={handleChange}
        className="form__checkbox"
        name="courier"
      />
      <Input
        type="text"
        value={login}
        onChange={handleChange}
        placeholder="login"
        className="form__input"
        name="login"
      />
      <Input
        type="text"
        value={email}
        onChange={handleChange}
        placeholder="email"
        className="form__input"
        name="email"
      />
      <br />
      <Input
        type="password"
        value={password}
        onChange={handleChange}
        placeholder="password"
        className="form__input"
        name="password"
      />
      <Input
        type="text"
        value={phone}
        onChange={handleChange}
        placeholder="phone"
        className="form__input"
        name="phone"
      />
      <br />
      <Input type="submit" value="Register" className="form__submit" name="submitButton" />
    </form>
  );
}
