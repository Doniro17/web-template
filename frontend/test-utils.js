import React from 'react'; // eslint-disable-line no-unused-vars

import { initFontAwesomeLibrary } from './src/utils';

import '@testing-library/jest-dom/extend-expect';

initFontAwesomeLibrary();

// re-export everything
export * from '@testing-library/react';
