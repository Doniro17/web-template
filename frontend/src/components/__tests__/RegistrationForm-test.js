import { render, fireEvent } from 'test-utils';

import RegistrationForm from '../RegistrationForm';

describe('LoginForm', () => {
  it('renders correctly', () => {
    const { container } = render(<RegistrationForm />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('Test checkbox status', () => {
    const { getByLabelText } = render(<RegistrationForm />);
    const checkbox = getByLabelText('Client');
    fireEvent.click(checkbox);
    expect(checkbox.checked).toBe(true);
  });
  it('Handle Login input correctly', () => {
    const { getByPlaceholderText } = render(<RegistrationForm />);
    const input = getByPlaceholderText('login');
    fireEvent.change(input, { target: { value: 'test' } });
    expect(input.value).toBe('test');
  });
});
