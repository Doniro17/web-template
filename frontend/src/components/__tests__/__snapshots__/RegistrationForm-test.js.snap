// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly 1`] = `
<form
  class="login__form"
>
  <h2
    class="form__title"
  >
    Registration
  </h2>
  <label
    class="container"
  >
    Client
    <input
      class="form__checkbox"
      name="client"
      type="checkbox"
      value="false"
    />
    <span
      class="checkmark"
    />
  </label>
  <label
    class="container"
  >
    Courier
    <input
      class="form__checkbox"
      name="courier"
      type="checkbox"
      value="false"
    />
    <span
      class="checkmark"
    />
  </label>
  <label>
    <input
      class="form__input"
      name="login"
      placeholder="login"
      type="text"
      value=""
    />
  </label>
  <label>
    <input
      class="form__input"
      name="email"
      placeholder="email"
      type="text"
      value=""
    />
  </label>
  <br />
  <label>
    <input
      class="form__input"
      name="password"
      placeholder="password"
      type="password"
      value=""
    />
  </label>
  <label>
    <input
      class="form__input"
      name="phone"
      placeholder="phone"
      type="text"
      value=""
    />
  </label>
  <br />
  <label>
    <input
      class="form__submit"
      name="submitButton"
      type="submit"
      value="Register"
    />
  </label>
</form>
`;
